test_jpeg/jpegenc/subblocks/RLE/RLECore/entropycoder.py

```diff
@@ -0,0 +1,46 @@
+""" This module takes a input and returns amplitude of the input
+    and number of bits required to store the input """
+
+from myhdl import always_seq, block
+from myhdl import intbv, ResetSignal, Signal
+from myhdl.conversion import analyze
+
+WIDTH_REG = 12
+SIZE = 4
```

Comment:
We talked about this before and I hadn't fully convinced you but you don't need defines (constants) like this in python/myhdl.  This is too much of Verilog/VHDL way of coding.  The blocks and functions are more dynamic, scalable, and module in python/myhdl you don't need functions and blocks that only work for specific widths.  These should be removed and the functions and blocks that use them should be modified to be more generic.

Slightly off topic, I don't understand why a separate repo is being used instead of your fork of test_jpeg?  It adds additional work (and tracking on my part) and postpones the integration, which is not desired.